src/utilities.ts

```diff
@@ -187,8 +187,22 @@ export function escapeRegExp(text: string) {
 	return text.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
 }
 
-export function EscapeHtml(text: string): string {
-	return new Option(text).innerHTML;
+export function EscapeHtml(unsafe: string): string {
+	// Step 1, create an element
+	const element = document.createElement('div');
+	// Step 2, safely add text to the element
+	element.textContent = unsafe;
+	/**
+	 * Step 3, let the browser handle turning most HTML characters into entities
+	 * Using innerText here returns a string that doesn't have HTML entities, so we use innerHTML to get entities.
+	 *
+	 * If a browser engine cannot do this, it means that setting Element.textContent is unsafe...
+	 */
+	const partiallySafe = element.innerHTML;
+	// Step 4, replace single and double quotes with entities so that the string can be added to attributes safely
+	const safe = partiallySafe.replace(/'/g, '&#039;').replace(/"/g, '&quot;').replace(/&(?![#\w]{2,};)/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
```

Comment:
any chance we can do something like this rather than a bunch of chained replaces that all need to parse the full string? 

https://github.com/kheina-com/frontend/blob/37f3a35ae41d7639d405e55e9a4a8c12526fcccc/src/utilities/markdown.js#L142-L148

I'm also not 100% sure what this is doing or why it's necessary `.replace(/&(?![#\w]{2,};)/g, '&amp;')` I don't think there's anything necessarily risky about ampersands in particular, but I'm not against escaping them, just in case.

I would also just return here rather than storing it in a variable and then escaping afterwards

Reply:
is this a blocker for merging, or can it be fixed later?

Reply:
> I'm also not 100% sure what this is doing or why it's necessary `.replace(/&(?![#\w]{2,};)/g, '&amp;')`

If you're confused why it's so complicated, it's so that the replacement doesn't take something like `&gt;` and spit out `&amp;gt;`. I'm ngl I also don't get what the threat with ampersands is, but [OSWAP](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#output-encoding-rules-summary) says to replace it with an HTML entity.

Reply:
<, >, and & are already sanitized to when you call 'const partiallySafe = element.innerHTML;' so those parts are unnecessary IMO 
![image](https://github.com/user-attachments/assets/f96b99b9-931c-413a-b1fa-e16209e713b6)

Reply:
As to why ampersands are escaped: https://stackoverflow.com/questions/39214564/why-ampersand-should-be-escaped-because-of-xss-injection

Reply:
> <, >, and & are already sanitized to when you call 'const partiallySafe = element.innerHTML;' so those parts are unnecessary IMO ![image](https://private-user-images.githubusercontent.com/78707594/348731685-f96b99b9-931c-413a-b1fa-e16209e713b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyMTAwMjksIm5iZiI6MTcyMTIwOTcyOSwicGF0aCI6Ii83ODcwNzU5NC8zNDg3MzE2ODUtZjk2Yjk5YjktOTMxYy00MTNhLWIxZmEtZTE2MjA5ZTcxM2I2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDA5NDg0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVkZTMwOTgyNzM2OTg2ZjJkNGQzMzkzNDlkNzI2ODIxZGM4YWU4ZDcwZGQ5MTExMzFlZjQ5OTFhMjM2ZWRjN2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GGk_pK6ZLv_A3rR1gJwBjVs8Gw-_QAJttLUUBAB8I3o)

They are, I mostly just added it to the replace chain so that Mozilla's reviewers can see the replacement happening explicitly? Might not be necessary?

Reply:
Ok removed the explicit <, >, and & replacements. I also validated that the OSWAP 5 are in fact replaced 
![image](https://github.com/user-attachments/assets/75aecf83-f242-41a4-a9a7-e915e1dc7989)